desktop/mgtk.s

```diff
@@ -2135,441 +2279,558 @@ ora_2_param_bytes:
         lda     (params_addr),y
         iny
         ora     (params_addr),y
-        sta     $B4
+        sta     vertices_count
         inc16   params_addr
         inc16   params_addr
         ldy     #$80
-L5379:  rts
+orts:   rts
 
 ;;; ============================================================
 ;;; InPoly
 
 InPolyImpl:
-        lda     #$80
-        bne     L5380
+        lda     #poly_oper_test
+        bne     PaintPolyImpl_entry2
 
 ;;; ============================================================
 ;;; PaintPoly
 
         ;; also called from the end of LineToImpl
 
-PaintPolyImpl:  lda     #$00
-L5380:  sta     $BA
+        num_maxima       := $AD
+        max_num_maxima   := 8
+
+        low_vertex       := $B0
+
+
+.proc PaintPolyImpl
+
+        lda     #poly_oper_paint
+entry2: sta     poly_oper
         ldx     #0
-        stx     $AD
+        stx     num_maxima
         jsr     ora_2_param_bytes
-L5389:  jsr     L52A1
-        bcs     L539D
-        ldx     $B0
-L5390:  jsr     L5354
-        bmi     L5389
-        jmp     L546F
-
-L5398:  exit_call $82
-
-L539D:  ldy     #1
-        sty     $AF
-        ldy     $AE
-        cpy     $B0
-        bne     L53A9
-        ldy     $B3
-L53A9:  dey
-        sty     $AB
+
+loop:   jsr     load_poly
+        bcs     process_poly
+        ldx     low_vertex
+next:   jsr     next_poly
+        bmi     loop
+
+        jmp     fill_polys
+
+bad_poly:
+        exit_call MGTK::error_bad_object
+.endproc
+
+
+        temp_yh        := $83
+        next_vertex    := $AA
+        current_vertex := $AC
+        loop_ctr       := $AF
+
+
+.proc process_poly
```

Comment:
If you've got the algorithm figured out, is it worth documenting?

Reply:
Still a work-in-progress, but yes as it comes clearer I'll put something.